### PR TITLE
feat(utilities): add omitKeysFromObject utility function

### PR DESCRIPTION
**Pull request by favna**
Ported from https://github.com/skyra-project/skyra/pull/2505#discussion_r1390246482 where I commented that the new utility function from Skyra would do well to be ported to Sapphire Utilities.

### CODE REVIEW

**Review by vladfrangu — DISMISSED** [review]

File: packages/utilities/src/lib/omitKeysFromObject.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Clones the source object using {@link deepClone} then deletes the specified keys with {@link Reflect.deleteProperty}
+ *
+ * @template TO - The object type.
+ * @template KTO - The keys of the object type.
+ *
+ * @param source - The input object.
+ * @param keys - The keys to omit from the object.
+ * @returns A new object without the specified keys.
+ */
+export function omitKeysFromObject<TO extends object, KTO extends keyof TO>(source: TO, ...keys: readonly KTO[]): Omit<TO, KTO> {
```
Comment (kyranet) [inline packages/utilities/src/lib/omitKeysFromObject.ts:13]: Extending the names would be more readable:
```suggestion
export function omitKeysFromObject<Object extends object, ObjectKeys extends keyof Object>(source: Object, ...keys: readonly ObjectKeys[]): Omit<Object, ObjectKeys> {
```

File: packages/utilities/tests/omitKeysFromObject.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { omitKeysFromObject } from '../src/lib/omitKeysFromObject';
+
+describe('omitKeysFromObject', () => {
```
Comment (kyranet) [inline packages/utilities/tests/omitKeysFromObject.test.ts:3]: There should be a test checking that `source` and `result` are different objects, and that `source` isn't mutated.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — DISMISSED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/utilities/src/lib/omitKeysFromObject.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * Clones the source object using {@link deepClone} then deletes the specified keys with {@link Reflect.deleteProperty}
+ *
+ * @template TO - The object type.
+ * @template KTO - The keys of the object type.
+ *
+ * @param source - The input object.
+ * @param keys - The keys to omit from the object.
+ * @returns A new object without the specified keys.
+ */
+export function omitKeysFromObject<Object extends object, ObjectKeys extends keyof Object>(
```
Comment (vladfrangu) [inline packages/utilities/src/lib/omitKeysFromObject.ts:13]: @kyranet shouldn't we be using {} instead of object?

**Review by vladfrangu — APPROVED** [review]

File: packages/utilities/src/lib/omitKeysFromObject.ts (hunk 3 above)
Comment (kyranet) [reply packages/utilities/src/lib/omitKeysFromObject.ts:13]: No, recent TS versions made `object` exclude `null`.

**Pull request merged**